bmad-core/tasks/discover-vcs.md

```diff
@@ -0,0 +1,187 @@
+# VCS Discovery Task
+
+## Purpose
+
+Identify and adapt to the team's version control system at project initialization.
+
+## Philosophy
+
+- Optimize for the 85-90% who use Git
+- Remain open for the 10-15% with special needs
+- Suggest best practices without forcing them
+
+## Task Instructions
+
+### Step 1: Initial Discovery
+
+```yaml
+elicit: true
+prompt: |
+  How does your team manage code versions?
+
+  1. Git with GitHub/GitLab/Bitbucket [MOST COMMON]
+  2. Git with corporate server
+  3. Other version control system
+  4. No version control needed
+  5. Custom/Complex setup
+
+  Select a number (1-5):
+```
+
+### Step 2A: Git-Based Workflows (Options 1-2)
+
+If user selects Git-based:
+
+```yaml
+elicit: true
+prompt: |
+  Which Git workflow best describes your team's approach?
```

Comment:
Would it make sense to have the agent trying to check the git history if we are in a brownfield and infer the workflow?

Reply:
That makes a lot of sense to me. 👍 Thank you. 

And like always, this is not set in stone; it can be changed/removed at a later date if it doesn't work.